engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schema/DefaultAvroSchemaEvolution.scala

```diff
@@ -47,16 +47,11 @@ class DefaultAvroSchemaEvolution extends AvroSchemaEvolution with DatumReaderWri
   /**
     * It's copy paste from AbstractKafkaAvroDeserializer#DeserializationContext.read with some modification.
     * We pass there record buffer data and schema which will be used to convert record.
-    *
-    * @param payload
-    * @param record
-    * @param writerSchema
-    * @param readerSchema
-    * @return
     */
-  protected def deserializePayloadToSchema(payload: Array[Byte], record: GenericContainer, writerSchema: Schema, readerSchema: Schema): Any = {
+  protected def deserializePayloadToSchema(payload: Array[Byte], writerSchema: Schema, readerSchema: Schema): Any = {
     try {
-      val reader = createDatumReader(record, writerSchema, readerSchema, useSchemaReflection = useSchemaReflection)
+      // We always want to create generic record at the end, because speecific can has other fields than expected
```

Comment:
I don't understand this case. What happens with evolution for specific record?

Reply:
Why we return there `Any` if we want always GenericData.Record here?